src/main/java/wooteco/subway/domain/Sections.java

```diff
@@ -0,0 +1,215 @@
+        int index = values.indexOf(section);
+        Section sectionToModify = new Section(section.getId(), section.getUpStation(), newSection.getUpStation(),
+            getNewDistance(section, newSection));
+        values.set(index, sectionToModify);
+        values.add(index + 1, newSection);
+        return sectionToModify;
+    }
+
+    private void validateAddable(Section section) {
+        List<Section> foundSections = findSectionsOverlapped(section);
+
+        if (isInvalid(section, foundSections)) {
+            throw new IllegalArgumentException("이미 포함된 두 역을 가진 Section 을 추가할 수 없습니다.");
+        }
+    }
+
+    private List<Section> findSectionsOverlapped(Section newSection) {
+        return values.stream()
+            .filter(section ->
+                section.contains(newSection.getDownStation()) || section.contains(newSection.getUpStation()))
+            .collect(Collectors.toList());
+    }
+
+    private boolean isInvalid(Section section, List<Section> foundSections) {
+        return hasSameStations(section) ||
+            isInvalidCountInMiddle(foundSections) || isSameWithDestinations(foundSections);
+    }
+
+    private boolean hasSameStations(Section newSection) {
+        return values.stream().anyMatch(section -> section.hasSameStations(newSection));
+    }
+
+    private boolean isInvalidCountInMiddle(List<Section> foundSections) {
+        int count = foundSections.size();
+        return count < 1 || count > 2;
+    }
+
+    private boolean isSameWithDestinations(List<Section> foundSections) {
+        return values.size() > 2 && foundSections.size() == 2 &&
+            foundSections.containsAll(List.of(values.get(0), values.get(values.size() - 1)));
+    }
+
+    public Station getUpDestination() {
+        return values.get(0).getUpStation();
+    }
+
+    public Station getDownDestination() {
+        return values.get(values.size() - 1).getDownStation();
+    }
+
+    public List<Section> getValues() {
+        return List.copyOf(values);
+    }
+
+    public Section delete(Station station) {
+        validateDeletable();
+
+        if (station.equals(getUpDestination())) {
+            return values.remove(0);
+        }
+
+        if (station.equals(getDownDestination())) {
+            return values.remove(values.size() - 1);
+        }
+
+        return mergeTwoSections(findSectionToDelete(station));
+    }
+
+    private void validateDeletable() {
+        if (values.size() <= 1) {
+            throw new IllegalArgumentException("구간이 하나인 경우 삭제할 수 없습니다.");
+        }
+    }
+
+    private Section findSectionToDelete(Station station) {
+        return values.stream()
+            .filter(it -> it.contains(station))
+            .findFirst()
+            .orElseThrow(() -> new IllegalArgumentException("삭제할 역이 존재하지 않습니다."));
+    }
+
+    private Section mergeTwoSections(Section section) {
+        int index = values.indexOf(section);
+        Section nextSection = findNextSection(section);
```

Comment:
`values.indexOf(section)`을 불필요하게 두 번 호출하게 되네요 👀

Reply:
세세하게 체크해주셔서 정말 감사드립니다🙏🙏. `Section`이 아니라 `index`를 받도록 하여 불필요한 호출을 수정하였습니다.

```suggestion
        int index = values.indexOf(section);
        Section nextSection = findNextSection(index);
        
        private Section findNextSection(int index) {
        return values.get(index + 1);
    }
```